src/conversion/azure/ip_converter.ts

```diff
@@ -1,88 +1,86 @@
-    const ip = this.parseIpAddress(input as T);
-    const key = store.getAlias(input.id);
+  if (subnetId) {
+    const subnet = store.getAlias(subnetId);
 
-    result.push({
-      key,
-      endpoint: true,
-      range: {
-        sourceIp: ip,
-      },
-      rules: this.parseSubnetRules(input as T, store),
+    rules.push({
+      destinationIp: subnet,
+      destination: subnet,
     });
-
-    return result;
   }
 
-  private parseSubnetRules(
-    input: T,
-    store: IEntityStore<AnyAzureObject>
-  ): ForwardRuleSpec[] {
-    const rules: ForwardRuleSpec[] = [];
-
-    const subnetId = this.parseSubnetId(input as T);
-
-    if (subnetId) {
-      const subnet = store.getAlias(subnetId);
-
-      rules.push({
-        destinationIp: subnet,
-        destination: subnet,
-      });
-    }
-
-    return rules;
-  }
-
-  protected abstract parseIpAddress(input: T): string;
-
-  protected abstract parseSubnetId(input: T): string | undefined;
+  return rules;
 }
 
-export class PublicIpConverter extends BaseIpConverter<AzurePublicIp> {
-  constructor() {
-    super('microsoft.network/publicipaddresses');
-  }
-
-  protected parseIpAddress(input: AzurePublicIp): string {
-    return input.properties.ipAddress;
-  }
-
-  protected parseSubnetId(input: AzurePublicIp): string | undefined {
-    return input.properties.subnet?.id;
-  }
+function parseNodeSpecs(
+  input: AnyAzureObject,
+  store: IEntityStore<AnyAzureObject>,
+  ip: string,
+  subnetRules: ForwardRuleSpec[]
+): NodeSpec[] {
+  const result: NodeSpec[] = [];
+  const key = store.getAlias(input.id);
+
+  result.push({
+    key,
+    endpoint: true,
+    range: {
+      sourceIp: ip,
+    },
+    rules: subnetRules,
+  });
+
+  return result;
 }
 
-export class LocalIpConverter extends BaseIpConverter<AzureLocalIP> {
-  constructor() {
-    super('Microsoft.Network/networkInterfaces/ipConfigurations');
-  }
-
-  protected parseIpAddress(input: AzureLocalIP): string {
-    return input.properties.privateIPAddress;
-  }
+function parseLocalIpSpec(
+  input: AnyAzureObject,
+  store: IEntityStore<AnyAzureObject>
+): NodeSpec[] {
+  const localIp = input as AzureLocalIP;
+  const ip = localIp.properties.privateIPAddress;
+  const rules = parseSubnetRules(localIp.properties.subnet?.id, store);
+  return parseNodeSpecs(input, store, ip, rules);
+}
 
-  protected parseSubnetId(input: AzureLocalIP): string | undefined {
-    return input.properties.subnet?.id;
-  }
+function parsePublicIpSpec(
+  input: AnyAzureObject,
+  store: IEntityStore<AnyAzureObject>
+): NodeSpec[] {
+  const publicIp = input as AzurePublicIp;
+  const ip = publicIp.properties.ipAddress;
+  const rules = parseSubnetRules(publicIp.properties.subnet?.id, store);
+  return parseNodeSpecs(input, store, ip, rules);
 }
+
+export const PublicIpConverter = {
+  supportedType: 'microsoft.network/publicipaddresses',
+  monikers: parseMonikers,
+  convert: parsePublicIpSpec,
+} as IAzureConverter;
```

Comment:
I think it is better to write
~~~
export const PublicIpConverter: IAzureConverter = { ... };
~~~

This should give you a more specific error message if there is a problem.